### ISSUE TITLE
Add support for Ahead-of-Time (AOT) compilation

### ISSUE THREAD

**Issue by raderio** [OpenFeign/feign#1082]
Will be great if Retrofit will generate interfaces implementation at compile time. Something like Micronaut does.

https://github.com/micronaut-projects/micronaut-core
Micronaut uses Ahead of Time (AOT) compilation to pre-compute your applications requirements at compile time. The result of this is significantly lower memory requirements, faster startup time, and reflection free framework infrastructure.

This will also make possible to better integration with GraalVM.

Another great example is Micronaut Data.icronaut Data is a database access toolkit that uses Ahead of Time (AoT) compilation to pre-compute queries for repository interfaces that are then executed by a thin, lightweight runtime layer.
https://github.com/micronaut-projects/micronaut-data
https://micronaut-projects.github.io/micronaut-data/latest/guide/

**Comment by velo** [comment]
Yeah, I'm working on it, some preparation work got merged and I expect to have this feature in a month or so...

Feel free to join the effort
https://github.com/velo/feign/tree/apt-generator

**Comment by lgklein** [comment]
Hello, any news about this or a way to make feign work with native images?

**Comment by GeorgeSalu** [comment]
Any updates ?

**Comment by velo** [comment]
> Any updates ?

I made a prototype, still waiting on more man power
